tox_packet/src/onion/onion_announce_request.rs

```diff
@@ -3,18 +3,21 @@
 
 use super::*;
 
-use std::convert::TryInto;
 use tox_binary_io::*;
 use tox_crypto::*;
 use crate::dht::*;
 
 use nom::{
     flat_map,
-    map_opt,
     number::complete::le_u64,
     combinator::{rest, rest_len},
     bytes::complete::take
 };
+use sha2::{Digest, Sha256};
+use sha2::digest::generic_array::typenum::marker_traits::Unsigned;
+
+/// The type of onion ping ID which is SHA256 hash.
+pub type PingId = [u8; <Sha256 as Digest>::OutputSize::USIZE];
```

Comment:
Looks like we can add `pub type Nonce = [u8; xsalsa20poly1305::NONCE_SIZE];` too. And all other bytish types.